### PR TITLE
Feature: add helpers to remove elements from FormArray

### PR DESCRIPTION
**Pull request by MarioArnt**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #40 

## What is the new behavior?

It is now possible to remove elements from `FormArray` that match a value or a predicate.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngneat/reactive-forms/src/lib/formArray.ts [hunk 1]
```diff
@@ -190,4 +190,18 @@ export class FormArray<T = any, E extends object = any> extends NgFormArray {
   setDisable(disable = true, opts?: ControlEventOptions) {
     disableControl(this, disable, opts);
   }
+
+  remove(value: T) {
+    this.removeWhen(v => v.value === value);
+  }
+
+  removeWhen(predicate: (element: AbstractControl<T>) => boolean) {
```
Comment (NetanelBasal) [inline projects/ngneat/reactive-forms/src/lib/formArray.ts:198]: - I would call it `removeIf`. 
- Why do you need the sorting functionality? Loop over the controls and remove them based on the predicate.

File: projects/ngneat/reactive-forms/src/lib/formArray.spec.ts [hunk 2]
```diff
@@ -212,4 +213,46 @@ describe('FormArray', () => {
     control.push(new FormControl('Phone'));
     expect(spy).toHaveBeenCalledWith(null);
   });
+  it('should remove', () => {
```
Comment (NetanelBasal) [inline projects/ngneat/reactive-forms/src/lib/formArray.spec.ts:216]: Add a space between each `it`.

File: projects/ngneat/reactive-forms/src/lib/formArray.spec.ts [hunk 3]
```diff
@@ -212,4 +213,46 @@ describe('FormArray', () => {
...
+    control.push(new FormControl('LastName'));
+    control.push(new FormControl('Phone'));
+    control.push(new FormControl('MiddleName'));
+    control.push(new FormControl('StreetNumber'));
+    control.push(new FormControl('StreetName'));
+    control.push(new FormControl('ZipCode'));
+    control.removeWhen(elt => elt.value.match(/Name$/) != null);
+    expect(control.getRawValue()).toEqual(['Phone', 'StreetNumber', 'ZipCode']);
+  });
+  it('should removeWhen', () => {
```
Comment (NetanelBasal) [inline projects/ngneat/reactive-forms/src/lib/formArray.spec.ts:240]: Don't duplicate descriptions.

File: projects/ngneat/reactive-forms/src/lib/formArray.spec.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { of, Subject } from 'rxjs';
```
Comment (NetanelBasal) [inline projects/ngneat/reactive-forms/src/lib/formArray.spec.ts:1]: Update the docs.

Comment (MarioArnt) [reply projects/ngneat/reactive-forms/src/lib/formArray.spec.ts:1]: done

File: projects/ngneat/reactive-forms/src/lib/formArray.spec.ts (hunk 2 above)
Comment (MarioArnt) [reply projects/ngneat/reactive-forms/src/lib/formArray.spec.ts:217]: done

File: projects/ngneat/reactive-forms/src/lib/formArray.ts (hunk 1 above)
Comment (MarioArnt) [reply projects/ngneat/reactive-forms/src/lib/formArray.ts]: The Angular `FormArray.prototype.removeAt` method is in-place.
So the array length will change during the loop when removing elements.
That's why it is required to remove elements from the higher index to the lower.

However good point, it's more straightforward and takes less time to iterate from the last to the first index and directly remove controls.

File: projects/ngneat/reactive-forms/src/lib/formArray.spec.ts (hunk 3 above)
Comment (MarioArnt) [reply projects/ngneat/reactive-forms/src/lib/formArray.spec.ts]: done

**Comment by MarioArnt** [comment]
> Add yourself to the contributors' list, please.

Wasn't here this weekend, thanks for merging and adding contributor commit :)

**Pull request merged**